routes/api_v1_0.py

```diff
@@ -207,27 +206,24 @@ def variables_query_v1_0():
 
     data = []
 
-    if 'vectors_only' in args:
-        for variable in config.vector_variables:
+    with open_dataset(config) as ds:
+        for v in ds.variables:
+            if config.variable[v.key].is_hidden:
+                continue
+
+            if ('3d_only' in args) and v.is_surface_only():
+                continue
+
+            if ('vectors_only' in args) and v.key not in config.vector_variables:
+                continue
+
             data.append({
-                'id': variable,
-                'value': config.variable[variable].name,
-                'scale': config.variable[variable].scale,
-                'interp':config.variable[variable].interpolation
+                'id': v.key,
+                'value': config.variable[v].name,
+                'scale': config.variable[v].scale,
+                'interp':config.variable[v].interpolation,
+                'two_dimensional': v.is_surface_only()
```

Comment:
API improvement: add extra information to API call so front-end can use it for filtering